packages/backend-modules/discussions/graphql/resolvers/Discussion/userCanComment.js

```diff
@@ -1,35 +1,33 @@
 const { Roles } = require('@orbiting/backend-modules-auth')
 const { hasUserActiveMembership } = require('@orbiting/backend-modules-utils')
 
-const DEFAULT_ROLES = ['member']
-
+/**
+ * User can comment when either:
+ * - a) it has a role which is allowed to comment (discussion.allowedRoles)
+ * - b) it has member role and either debater role or an active membership
+ */
 module.exports = async (discussion, _, context) => {
   const { closed } = discussion
   const { user, pgdb } = context
 
+  // User can not comment, if discussion is closed
   if (closed) {
     return false
   }
 
-  const additionalAllowedRoles = discussion.allowedRoles.filter(
-    (role) => ![DEFAULT_ROLES].includes(role),
-  )
-  const isInAllowedRoles =
-    additionalAllowedRoles.length > 0 &&
-    Roles.userIsInRoles(user, discussion.allowedRoles)
-
-  if (isInAllowedRoles) {
+  // a) User can comment, if it has some of the allowed roles but member
+  const allowedRoles = discussion.allowedRoles.filter((r) => r !== 'member')
+  if (Roles.userIsInRoles(user, allowedRoles)) {
     return true
   }
 
+  // b) User can comment, if discussion allows member role, user has member role
+  // and either debater role or an active membership.
+  const isMemberAllowed = !!discussion.allowedRoles.find((r) => r === 'member')
```

Comment:
Are there discussions where this isn't the case? Or do you mean that the discussion explicitly allows everyone with a member role (with or without active membership) to participate?

Reply:
Code allows for `discussions.allowedRoles` to be an empty array. If so, user should not be able to comment.

Reply:
Nit: IMHO this could be written more clearly as `discussion.allowedRoles.some((r) => r === 'member'` instead of (mis-) using `find` and the double-falsification `!!`

Reply:
True!!